controllers/nodefeaturediscovery_resources.go

```diff
@@ -205,3 +208,263 @@ func panicIfError(err error) {
+func (r *NodeFeatureDiscoveryReconciler) getClusterRole(ctx context.Context, namespace string, name string) (*rbacv1.ClusterRole, error) {
+	cr := &rbacv1.ClusterRole{}
+	err := r.Get(ctx, client.ObjectKey{Namespace: namespace, Name: name}, cr)
+	return cr, err
+}
+
+// getClusterRoleBinding gets one of the NFD Operator's ClusterRoleBindings
+func (r *NodeFeatureDiscoveryReconciler) getClusterRoleBinding(ctx context.Context, namespace string, name string) (*rbacv1.ClusterRoleBinding, error) {
+	crb := &rbacv1.ClusterRoleBinding{}
+	err := r.Get(ctx, client.ObjectKey{Namespace: namespace, Name: name}, crb)
+	return crb, err
+}
+
+// getSecurityContextConstraints gets one of the NFD Operator's SecurityContextConstraints
+func (r *NodeFeatureDiscoveryReconciler) getSecurityContextConstraints(ctx context.Context, namespace string, name string) (*secv1.SecurityContextConstraints, error) {
+	scc := &secv1.SecurityContextConstraints{}
+	err := r.Get(ctx, client.ObjectKey{Namespace: namespace, Name: name}, scc)
+	return scc, err
+}
+
+// deleteServiceAccount deletes one of the NFD Operator's ServiceAccounts
+func (r *NodeFeatureDiscoveryReconciler) deleteServiceAccount(ctx context.Context, namespace string, name string) error {
+
+	// Attempt to get the existing ServiceAccount from the reconciler
+	sa, err := r.getServiceAccount(ctx, namespace, name)
+
+	// If the resource was not found, then do not return an
+	// error because this means the resource has already
+	// been deleted
+	if k8serrors.IsNotFound(err) {
+		return nil
+	}
+
+	// If some other error occurred when trying to get the
+	// resource, then return that error
+	if err != nil {
+		return err
+	}
+
+	// ...otherwise, delete it
+	return r.Delete(context.TODO(), sa)
+}
+
+// deleteDaemonSet deletes the NFD Operator's DaemonSet (worker or master)
+func (r *NodeFeatureDiscoveryReconciler) deleteDaemonSet(ctx context.Context, namespace string, name string) error {
+
+	// Attempt to get the existing DaemonSet from the reconciler
+	ds, err := r.getDaemonSet(ctx, namespace, name)
+
+	// If the resource was not found, then do not return an
+	// error because this means the resource has already
+	// been deleted
+	if k8serrors.IsNotFound(err) {
+		return nil
+	}
+
+	// If some other error occurred when trying to get the
+	// resource, then return that error
+	if err != nil {
+		return err
+	}
+
+	// ...otherwise, delete it
+	return r.Delete(context.TODO(), ds)
+}
+
+/*
+// deleteWorkerConfig deletes the NFD Operator's worker ConfigMap
+func (r *NodeFeatureDiscoveryReconciler) deleteWorkerConfig(n NFD) error {
+
+	// Attempt to get the existing ConfigMap from the reconciler
+	wc := n.ins.Spec.WorkerConfig
+
+	// Get the config data so we can see if it's empty
+	cd := wc.ConfigData
+
+	// If 'cd' is empty, then the resource does not exist, so it cannot
+	// be deleted
+	if cd == "" {
+		return nil
+	}
+
+	return r.Delete(context.TODO(), wc)
+}
+*/
```

Comment:
?

Reply:
Ah, yep. Good catch. Let me delete that.